spec/models/service_terraform_template_spec.rb

```diff
@@ -326,6 +326,62 @@
     end
   end
 
+  describe "#execute_async" do
+    [
+      ResourceAction::PROVISION,
+      ResourceAction::RECONFIGURE,
+      ResourceAction::RETIREMENT
+    ].each do |action|
+      describe "with #{action} action" do
+        let!(:service) do
+          FactoryBot.create(:service_terraform_template).tap do |s|
+            s.add_resource!(stack1, :name => ResourceAction::PROVISION)
+            case action
+            when ResourceAction::RECONFIGURE
+              s.add_resource!(stack2, :name => ResourceAction::RECONFIGURE)
+            when ResourceAction::RETIREMENT
+              s.add_resource!(stack2, :name => ResourceAction::RECONFIGURE)
+              s.add_resource!(stack3, :name => ResourceAction::RETIREMENT)
+            end
```

Comment:
Do we need these stacks to be added as resources for anything that is being tested here?